src/ui/TableOfContent.tsx

```diff
@@ -142,3 +93,56 @@ export default function TableOfContent({
     </Menu>
   );
 }
+
+type TocItemProps = React.ComponentPropsWithoutRef<typeof MenuItem> & {
+  href: string;
+  title: string | undefined;
+  isActive: boolean;
+  onClick: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+};
+
+const TocItem = (props: TocItemProps) => {
+  const { href, title, isActive, ...rest } = props;
+
+  const bgColor = useColorModeValue('ui.white', 'ui.black', 'ui.sepia');
+  const color = useColorModeValue('ui.black', 'ui.white', 'ui.black');
+  const borderColor = useColorModeValue(
+    'ui.gray.medium',
+    'gray.500',
+    'yellow.600'
+  );
+
+  const activeColor = 'ui.black';
+  const activeBgColor = 'ui.gray.light-cool';
+
+  const _hover = {
+    textDecoration: 'none',
+    background: isActive ? 'ui.black' : 'ui.gray.x-dark',
+    color: 'ui.white',
+  };
+
+  const _focus = {
+    ..._hover,
+    boxShadow: 'none',
+  };
+
+  const styles = {
+    background: isActive ? activeBgColor : bgColor,
+    color: isActive ? activeColor : color,
+    borderBottom: `1px solid`,
+    borderColor: borderColor,
+    py: 3,
+    _hover,
+    _focus,
+  };
+
+  return (
+    <MenuItem as={Link} href={href} {...styles} {...rest}>
+      <RenderHtml title={title ?? ''} />
+    </MenuItem>
+  );
+};
+
+const RenderHtml = ({ title }: { title: string }) => (
+  <span dangerouslySetInnerHTML={{ __html: title }}></span>
+);
```

Comment:
I had to create a separate component for the HTML component because I was getting [Can only set one of `children` or `props.dangerouslySetInnerHTML`](https://stackoverflow.com/a/39910404/8606992) error.